test/LogicalImagesTests.cpp

```diff
@@ -15,3 +27,55 @@ TEST_F(LogicalImagesTests, assertThatJSONParsingLibraryWorks)
 
     ASSERT_EQ(j.dump(), "{\"pi\":3.14}");
 }
+
+TEST_F(LogicalImagesTests, checkFirstConstructorNoexception)
+{
+    std::string path = "../test/ReadMeBoardTestFile.json";
+    JSONFileReader jfr(path);
+    auto b = std::make_shared<Board>(jfr.readRowSize(),
+                                     jfr.readColSize(),
+                                     jfr.readCluesRows(),
+                                     jfr.readCluesCols());
+    auto a = std::make_shared<AlgorithmA>();
+
+    LogicalImages li(b, a);
+}
+
+TEST_F(LogicalImagesTests, checkSecondConstructorNoexception)
+{
+    std::string path = "../test/ReadMeBoardTestFile.json";
+    auto a = std::make_shared<AlgorithmA>();
+
+    LogicalImages li(path, a);
+}
+
+TEST_F(LogicalImagesTests, checkSecondConstructorExceptions)
+{
+    std::string path;
+    auto a = std::make_shared<AlgorithmA>();
+
+    path = "../test/JSONFileReaderWrongColSizeTestFile.json";
+    ASSERT_THROW(LogicalImages li1(path, a), InvalidDimensions);
+
+    path = "../test/JSONFileReaderWrongRowSizeTestFile.json";
+    ASSERT_THROW(LogicalImages li2(path, a), InvalidDimensions);
+
+    path = "../test/JSONFileReaderNumberOfLinesAndColsNoMatch.json";
+    ASSERT_THROW(LogicalImages li3(path, a), InvalidDimensions);
+
+    path = "../test/JSONFileReaderNumberOfLinesAndRowsNoMatch.json";
+    ASSERT_THROW(LogicalImages li4(path, a), InvalidDimensions);
+}
+
+TEST_F(LogicalImagesTests, checkSolveAlgorithmA)
+{
+    std::string path = "../test/ReadMeBoardTestFile.json";
+    std::string solution = "Puzzle solved!";
+    auto a = std::make_shared<AlgorithmA>();
+    testing::internal::CaptureStdout();
+
+    LogicalImages li(path, a);
+    li.solve();
```

Comment:
Using cout in tests is not advised. You can return some value from solve method and check it.

Reply:
How then solve problem of code coverage with tests? Or can't it be used to suppress error msgs from exceptions, so they won't appear on screen when testing?
This functions is purly to run solving algorithm and print solution if found.

Reply:
You can ignore coverage for now if you know that there will be some tests implemented later, which will cover this implementation. 
Coverage is useful, but do not blindly try to achieve 100%. Usually, it means that everything is done for statistics, not real work ;)